lib/countly.js

```diff
@@ -2002,6 +2025,46 @@
            loader.style.display = "none"; 
         }
     }
+
+    /**
+    * Show feedback popup by passed widget ids array
+    * @param {object=} configuration - required - includes "widgets" property as string array of widgets
+    * example configuration: {"widgets":["5b21581b967c4850a7818617"]}
+    **/
+    Countly.enable_feedback = function(params){
+        // inject feedback styles
+        loadCSS(Countly.url + '/star-rating/stylesheets/countly-feedback-web.css');
+        // get enable widgets by app_key
+        // define xhr object
+        var enableWidgets = params.widgets;
+
+        if (enableWidgets.length > 0) {
+            
+            document.body.innerHTML += '<div id="cfbg"></div>';
+
+            sendXmlHttpRequest(Countly.url + '/o/feedback/multiple-widgets-by-id', {app_key:Countly.app_key, widgets:JSON.stringify(params.widgets)}, function(err, params, responseText){
+                if(err){
+                    log("Errors occurred:", err);
+                }
+                // widget object
+                var widgets = JSON.parse(responseText);
+                for (var i = 0; i < widgets.length; i++) {
+                    var pages = JSON.parse(widgets[i].target_pages);
+                    for (var j = 0; j < pages.length; j++) {
+                        if (pages[j] === window.location.pathname && !widgets[i].hide_sticker) document.body.innerHTML += '<div style="color:'+((widgets[i].trigger_font_color < 7) ? '#'+widgets[i].trigger_font_color : widgets[i].trigger_font_color)+";background-color:"+((widgets[i].trigger_bg_color.length < 7) ? '#'+widgets[i].trigger_bg_color : widgets[i].trigger_bg_color)+'" class="countly-feedback-sticker '+widgets[i].trigger_position+'" id="countly-feedback-sticker-'+widgets[i]._id+'">'+widgets[i].trigger_button_text+'</div><div class="countly-iframe-wrapper" id="countly-iframe-wrapper-'+widgets[i]._id+'"><span class="countly-feedback-close-icon" id="countly-feedback-close-icon-'+widgets[i]._id+'">×</span><iframe name="countly-feedback-iframe" id="countly-feedback-iframe" src="'+Countly.url+"/feedback?widget_id="+widgets[i]._id+"&app_key="+Countly.app_key+'&device_id='+Countly.device_id+'"></iframe></div>';
+                    } 
+                    for (var k = 0; k < pages.length; k++) {
+                        if (pages[k] === window.location.pathname && !widgets[i].hide_sticker) {
+                            add_event(document.getElementById('countly-feedback-sticker-'+widgets[i]._id), 'click', function(){document.getElementById('countly-iframe-wrapper-'+widgets[i]._id).style.display = "block";document.getElementById('cfbg').style.display = "block";});
```

Comment:
Here in the add_event callback **widgets[i]._id** will be incorrect, cause loop already ended, so it will be equal to pages.length always.